src/runtime/ThreadLocal.cpp

```diff
@@ -131,6 +142,45 @@ void ThreadLocal::initialize()
     // Heap is initialized for each thread
     Heap::initialize();
 
+    // g_stackLimit
+#if defined(OS_WINDOWS)
+    ULONG_PTR low, high;
+    GetCurrentThreadStackLimits(&low, &high);
+    void* stackStartAddress = reinterpret_cast<void*>(low);
+    void* stackEndAddress = reinterpret_cast<void*>(high);
+    size_t stackSize = reinterpret_cast<size_t>(stackEndAddress) - reinterpret_cast<size_t>(stackStartAddress);
+#else
+    void* stackStartAddress;
+    void* stackEndAddress;
+    size_t stackSize;
+#if defined(OS_DARWIN)
+    stackStartAddress = pthread_get_stackaddr_np(pthread_self());
+    stackSize = pthread_get_stacksize_np(pthread_self());
+#else
+    pthread_attr_t attr;
+    pthread_getattr_np(pthread_self(), &attr);
+    pthread_attr_getstack(&attr, &stackStartAddress, &stackSize);
+    pthread_attr_destroy(&attr);
+#endif
+#ifdef STACK_GROWS_DOWN
+    stackEndAddress = (char*)stackStartAddress - stackSize;
+#if defined(OS_DARWIN)
+    std::swap(stackStartAddress, stackEndAddress);
+#endif
+#else
+    stackEndAddress = (char*)stackStartAddress + stackSize;
+#endif
+#endif
+    stackSize = std::min(stackSize, (size_t)STACK_USAGE_LIMIT);
+
+#ifdef STACK_GROWS_DOWN
+    UNUSED_VARIABLE(stackEndAddress);
+    g_stackLimit = reinterpret_cast<size_t>(stackStartAddress) + STACK_FREESPACE_FROM_LIMIT;
```

Comment:
This limit means that we only use `STACK_FREESPACE_FROM_LIMIT` size for stack which is 256 KB,
Am I correct?

Reply:
If real stack size is 8MB, we will use (8MB-STACK_FREESPACE_FROM_LIMIT).
remained space used by computing loc when stackoverflow is occured

Reply:
I see!